pkg/controller/config/config_controller_test.go

```diff
@@ -664,3 +676,35 @@ func unstructuredFor(gvk schema.GroupVersionKind, name string) *unstructured.Uns
 type testExpectations interface {
 	IsExpecting(gvk schema.GroupVersionKind, nsName types.NamespacedName) bool
 }
+
+func ensureCreated(ctx context.Context, c client.Client, toCreate func() client.Object) func() error {
+	return func() error {
+		instance := toCreate()
+		key := client.ObjectKey{Namespace: instance.GetNamespace(), Name: instance.GetName()}
+		gvk := instance.GetObjectKind().GroupVersionKind()
+
+		u := &unstructured.Unstructured{}
+		u.SetGroupVersionKind(gvk)
+
+		err := c.Get(ctx, key, u)
+		if err == nil {
+			if !u.GetDeletionTimestamp().IsZero() {
+				return fmt.Errorf("waiting for deletion: %v %v", gvk, key)
+			}
+
+			err = c.Delete(ctx, instance)
```

Comment:
Is there some reason we want to proactively delete the object if it exists? `ensureCreated()` reads to me like "create if missing"

Reply:
Because if we don't delete the object, then the previous functionality being tested may carry over state. Since we obviously want tests to be independent, we do not want state to carry over between tests.

Reply:
I wonder if this logic would be simpler if we separated two distinct ideas:

1) Creating a clean slate (`ensureClean()`)
1) Applying our new resource

It's taken quite a bit of explaining for me to understand `ensureCreated()`, which strikes me as a sign that it could be simpler or more direct.

Reply:
Done. Separated into `ensureDeleted` and `ensureCreated`.